packages/binding-http/src/http-server.ts

```diff
@@ -298,28 +299,25 @@ export default class HttpServer implements ProtocolServer {
     public destroy(thingId: string): Promise<boolean> {
         debug(`HttpServer on port ${this.getPort()} destroying thingId '${thingId}'`);
         return new Promise<boolean>((resolve, reject) => {
-            let removedThing: ExposedThing | undefined;
-            for (const name of Array.from(this.things.keys())) {
-                const expThing = this.things.get(name);
-                if (expThing?.id === thingId) {
-                    this.things.delete(name);
-                    removedThing = expThing;
+            for (const [key, exposedThing] of this.things.entries()) {
+                if (exposedThing?.id === thingId) {
+                    this.things.delete(key);
+                    info(`HttpServer successfully destroyed '${exposedThing.title}'`);
+                    resolve(true);
+                    return;
                 }
             }
-            if (removedThing) {
-                info(`HttpServer successfully destroyed '${removedThing.title}'`);
-            } else {
-                info(`HttpServer failed to destroy thing with thingId '${thingId}'`);
-            }
-            resolve(removedThing !== undefined);
+
+            info(`HttpServer failed to destroy thing with thingId '${thingId}'`);
+            resolve(false);
         });
     }
 
     private addUrlRewriteEndpoints(form: TD.Form, forms: Array<TD.Form>): void {
-        if (this.urlRewrite) {
-            for (const inUri in this.urlRewrite) {
-                const toUri = this.urlRewrite[inUri];
-                if (form.href.endsWith(toUri)) {
+        if (this.urlRewrite != null) {
+            for (const [inUri, toUri] of Object.entries(this.urlRewrite)) {
+                // TODO: Without === true, this is currently an eslint false positiv
+                if (form.href.endsWith(toUri) === true) {
```

Comment:
Should we just disable the rule? or is it maybe that endsWith can result in undefined? 🤔

Reply:
I think this issue as well as others should get resolved by #1103, I will therefore remove the `=== true` check as well as the TODO :)

Reply:
I now managed to get rid of the `=== true` approach using some explicit typing :)